### PR TITLE
Changes to metazoa/nonvert reference sets - First time update

### PR DESCRIPTION
**Pull request by ens-LCampbell**
## Description

PR includes all new metazoan (non-vertebrate specific clades) reference sets outlined in [QRP meeting (20-07-22)](https://www.ebi.ac.uk/seqdb/confluence/display/ENS/2022-07-20+-+QRP+Weekly+meeting) where we first described and agreed on the four sets ("hexapoda", "arthropoda", "protostomia" and "metazoa"). 

**Related JIRA tickets:**
No Jira ticket related to this work

## Overview of changes
Three new sets and one alteration have been made to homology reference sets. It was pointed out that ref sets containing overlapping species can cause issues when said species have assembly/annotation updates. Care was taken to minimise overlapping species where possible but limited to groups in which we have many species to choose from. 

_#### Change ("Shared metazoa")_ 
**UPDATE (30-08-22). No longer part of this PR.**
 

#### Change 1
Replaced "hexapoda" set as it was defined, with updated set of insect species (mostly Neoptera orders). Name hexapoda was maintained.  

#### Change 2
Addition of "arthropoda", "protostomia" and "metazoa" reference sets. 

## Testing
Not tested yet. 

## Notes
The following four species are being transferred from Wormbase/WBParasite and are not with Metazoa yet, work in progress. Production names may change or need to be adjusted on our side or in ref sets. 

- Clonorchis sinensis	Chinese liver fluke	GCA_003604175.2	clonorchis_sinensis_gca003604175v2
- Echinococcus multilocularis	Tapeworm	GCA_000469725.3	echinococcus_multilocularis_gca000469725v3
- Schmidtea mediterranea	Planarian worm	GCA_002600895.1	schmidtea_mediterranea_gca002600895v1
- Haemonchus contortus	Barber pole worm	GCA_000469685.2	haemonchus_contortus_gca000469685v2

---

## PR review checklist

- [ ] Is the PR against an appropriate branch?
- [ ] Does the code adhere to coding guidelines?
- [ ] Does the code do what it claims to do?
- [ ] Is the code readable? Is it appropriately documented?
- [ ] Is the logic in the correct place?
- [ ] Was the code tested appropriately? Are there unit tests? Are unit tests self-contained and non-redundant?
- [ ] Did Travis CI pass for the code in the PR? Is Codecov acceptable based on the included/updated unit tests?
- [ ] Will the new code fail gracefully?
- [ ] Does the code follow good practice for writing performant code (e.g. using a database transaction rather than repeated queries outside of a transaction)?
- [ ] Does it bring in an unnecessary dependency?
- [ ] If you are reviewing a new analysis, is it future-proof and pluggable?
- [ ] Does the PR meet agile guidelines?

### CODE REVIEW

**Review by CristiGuijarro — COMMENTED** [review]
Except for the above comment on registry configuration, and following Travis's checks (in which conf/references/allowed_species.json needs to be updated with the inclusion of `salpingoeca_rosetta_gca_000188695`, `sphaeroforma_arctica_jp610_gca_001186125` and `capsaspora_owczarzaki_atcc_30864_gca_000151315`, I have checked that the new collection names work for the `scientific_name` in the `ncbi_taxonomy`. The distribution and selection of available references across clades looks more than reasonable, particularly with regards to the support from Wormbase and communication between Metazoa team and other experts in the field. This will be a welcome addition to the Metazoa and non-vert community and a decent improvement on the previous `hexapoda` collection :+1: 
Will approve once `mlss_conf.xml` and `allowed_species.json` are appropriately updated :relieved:

**Comment by CristiGuijarro** [comment]
(Travis on python linter failed tests can be ignored, this will be fixed on the `feature/homology_annotation` branch soon).

File: conf/references/production_reg_conf.pl
Comment (CristiGuijarro) [inline conf/references/production_reg_conf.pl:54]:
```suggestion
    -host   => 'mysql-ens-mirror-3',
```

File: conf/references/production_reg_conf.pl
Comment (CristiGuijarro) [inline conf/references/production_reg_conf.pl:61]:
```suggestion
    -host   => 'mysql-ens-mirror-3',
```

File: conf/references/production_reg_conf.pl
Comment (CristiGuijarro) [inline conf/references/production_reg_conf.pl:58]: `${curr_eg_release}` needs to be defined, preferably around line 35

And these collection dbs need to be included in the section for `PREVIOUS CORE DATABASES` (below line 68) with `${prev_eg_release}_${prev_release}` - so `${prev_eg_release}` will also need defining

**Review by CristiGuijarro — COMMENTED** [review]
A few changes needed in the `production_reg_conf.pl` and just missing the 3 new protists in the `allowed_species.json` - but nearly there, thanks!

File: conf/references/production_reg_conf.pl
Comment (JAlvarezJarreta) [reply conf/references/production_reg_conf.pl:58]: :facepalm: My fault, I forgot completely about that part when helping Lahcen to define these :sweat_smile:

Comment (vsitnik) [reply conf/references/production_reg_conf.pl:58]: as for `${curr_eg_release}`, if we're talking about the released data, `curr_eg_release = curr_release - 53`. should work

**Comment by CristiGuijarro** [comment]
Still missing:
```
  "salpingoeca_rosetta_gca_000188695",
  "sphaeroforma_arctica_jp610_gca_001186125",
   "capsaspora_owczarzaki_atcc_30864_gca_000151315"
 ```
 In https://github.com/Ensembl/ensembl-compara/blob/feature/homology_annotation/conf/references/allowed_species.json

File: conf/references/mlss_conf.xml
Comment (dthybert) [reply conf/references/mlss_conf.xml:42]: As discussed with Jorge and then Lahcen, the shared_metazoa should be removed and the shared used instead

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:272]: replace by "shared"

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:276]: could you please add a comment defining its clade (is Entognatha the right clade for this species?)


--

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:407]: need to be changed to shared

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:427]: I would add the mouse rather than the pig here. Again this will add very little diversity when present with the human. However, the mouse is one of the two best-annotated genomes (structural and functional) in eukaryotes so it would make sense to have it here.

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:426]: since the zebrafish is in the shared group would not it be better to add a species in clades with longer branches than vertebrates? At the metazoa level for the vertebrate, you already have the human, the chicken,  the xenopus, and the zebrafish which is already a good representation of the vertebrates (with the perspective of all metazoa). oryzias_latipes will add very little additional diversity at the metazoa level when the zebrafish is present also oryzias_latipes has a much worst annotation than the zebrafish

File: conf/references/mlss_conf.xml
Comment (dthybert) [inline conf/references/mlss_conf.xml:357]: change to shared

**Review by dthybert — CHANGES_REQUESTED** [review]

File: conf/references/mlss_conf.xml
Comment (ens-LCampbell) [reply conf/references/mlss_conf.xml:426]: Hi David do you have any suggestions for the rice fish replacement ? I tried to include Monotreme reps in the form of platypus to expand the diversity but that choice didn't sit well. Perhaps a reptile would be useful to include within verts instead of rice fish

File: conf/references/mlss_conf.xml
Comment (ens-LCampbell) [reply conf/references/mlss_conf.xml:427]: The reason for choosing Pig was its annotation recently had a big update and likely no further updates in the coming year at least. Avoiding mouse due to the impression of being updated more regularly and therefore would cause increased recomputes ?

File: conf/references/mlss_conf.xml
Comment (ens-LCampbell) [reply conf/references/mlss_conf.xml:276]: Collembola I think is best. Entognatha is also correct but not accepted taxonomic clade in NCBI (missing taxon id)

**Comment by ens-LCampbell** [comment]
Since having a final decision on use or not of default species set, I have now revert back to the original set of nine default species. 

I've also removed any mention of the suggested protist species and their collection db information from the production_ref_conf.pl.

**Review by dthybert — CHANGES_REQUESTED** [review]

**Comment by ens-LCampbell** [comment]
Hi @dthybert. I've updated the mlss_conf to reflect your suggestions regards the previously added vert species. I appreciate the input and have made two slight adjusts based on your input. Removal of pig and rice fish, and reintroduction of two species of protists within the metazoa ref set. 

I've also updated the allowed_species.json to reflect the species changes introduced across all ref set updates/additions. 

Best, LC

**Review by dthybert — APPROVED** [review]

**Comment by JAlvarezJarreta** [comment]
@dthybert could you merge this PR, please? We do not want to miss having these new reference sets when e108 is released :smile_cat:

**Pull request merged**
